azurerm/internal/services/storage/storage_share_data_source.go

```diff
@@ -0,0 +1,129 @@
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+
+			"storage_account_name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+
+			"metadata": MetaDataComputedSchema(),
+
+			"acl": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 64),
+						},
+						"access_policy": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"start": {
+										Type:         pluginsdk.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"expiry": {
+										Type:         pluginsdk.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"permissions": {
+										Type:         pluginsdk.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"quota": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"resource_manager_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceStorageShareRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	storageClient := meta.(*clients.Client).Storage
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	shareName := d.Get("name").(string)
+	accountName := d.Get("storage_account_name").(string)
+
+	account, err := storageClient.FindAccount(ctx, accountName)
+	if err != nil {
+		return fmt.Errorf("error retrieving Account %q for Share %q: %s", accountName, shareName, err)
+	}
+	if account == nil {
+		return fmt.Errorf("unable to locate Account %q for Share %q", accountName, shareName)
+	}
+
+	client, err := storageClient.FileSharesClient(ctx, *account)
+	if err != nil {
+		return fmt.Errorf("building FileShares Client for Storage Account %q (Resource Group %q): %s", accountName, account.ResourceGroup, err)
+	}
+
+	id := parse.NewStorageShareDataPlaneId(accountName, storageClient.Environment.StorageEndpointSuffix, shareName).ID()
+	d.SetId(id)
+	props, err := client.Get(ctx, account.ResourceGroup, accountName, shareName)
+	if err != nil {
+		return fmt.Errorf("retrieving Share %q (Account %q / Resource Group %q): %s", shareName, accountName, account.ResourceGroup, err)
+	}
+	if props == nil {
+		return fmt.Errorf("share %q was not found in Account %q / Resource Group %q", shareName, accountName, account.ResourceGroup)
+	}
+
+	d.Set("name", shareName)
+	d.Set("storage_account_name", accountName)
+	d.Set("quota", props.QuotaGB)
+	d.Set("acl", flattenStorageShareACLs(props.ACLs))
```

Comment:
since this is a complex object can we make this:

```suggestion
	if err := d.Set("acl", flattenStorageShareACLs(props.ACLs)); err != nil {
		return fmt.Errorf("setting `acl`: %+v", err)
	}
```